hasktorch/src/Torch/Utils/Image.hs

```diff
@@ -31,27 +31,29 @@ readImage file = do
   dynamic_img <- I.readImage file
   case dynamic_img of
     Left err -> return $ Left err
-    Right img' -> do
-      case img' of
-        I.ImageY8 (I.Image width height vec) -> createTensor width height 1 D.UInt8 vec
-        I.ImageYA8 (I.Image width height vec) -> createTensor width height 2 D.UInt8 vec
-        I.ImageRGB8 (I.Image width height vec) -> createTensor width height 3 D.UInt8 vec
-        I.ImageRGBA8 (I.Image width height vec) -> createTensor width height 4 D.UInt8 vec
-        I.ImageYCbCr8 (I.Image width height vec) -> createTensor width height 3 D.UInt8 vec
-        I.ImageCMYK8 (I.Image width height vec) -> createTensor width height 4 D.UInt8 vec
-        _ -> return $ Left "Unsupported format"
+    Right img' -> fromDynImage img'
+
+fromDynImage :: I.DynamicImage -> IO (Either String D.Tensor)
+fromDynImage image = case image of
+  I.ImageY8 (I.Image width height vec) -> createTensor width height 1 D.UInt8 vec
+  I.ImageYA8 (I.Image width height vec) -> createTensor width height 2 D.UInt8 vec
+  I.ImageRGB8 (I.Image width height vec) -> createTensor width height 3 D.UInt8 vec
+  I.ImageRGBA8 (I.Image width height vec) -> createTensor width height 4 D.UInt8 vec
+  I.ImageYCbCr8 (I.Image width height vec) -> createTensor width height 3 D.UInt8 vec
+  I.ImageCMYK8 (I.Image width height vec) -> createTensor width height 4 D.UInt8 vec
+  _ -> return $ Left "Unsupported format, convert to 8 bit Image first"
```

Comment:
actually, if we use the `convertRGB8` function, we could probably get rid of the `Either`?

Reply:
If there is no broken implementation making the image-data, we can erase it

Reply:
I did it.
https://github.com/hasktorch/hasktorch/blob/85ea2c9/hasktorch/src/Torch/Vision.hs#L95-L110